vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImplBase.java

```diff
@@ -67,7 +67,19 @@ protected boolean iterateNext() {
     boolean failed = failed();
     while (iter.hasNext()) {
       RouteImpl route = iter.next();
-      if (route.matches(this, mountPoint(), failed)) {
+      boolean matches;
+
+      try {
+        matches = route.matches(this, mountPoint(), failed);
+      } catch (IllegalArgumentException e) {
+        failed = true;
+        matches = false;
+        // Failure in handling failure!
+        if (log.isTraceEnabled()) log.trace("Failure in request match", e);
+        unhandledFailure(400, e, route.router());
```

Comment:
This is tricky indeed. If there is an `IllegalArgumentException` is should be thrown during the match since the params are loaders and possibly the path normalization is triggered. However if after the match this exception is thrown indeed a 400 status code is not correct anymore but because params and path normalization are lazy evaluated we cannot handle it correctly.